tests/integration/test_split.py

```diff
@@ -209,97 +224,81 @@ def compare_array(a, b):
-
-                # out index
-                indices_arr.append(
-                    list(
-                        range(
-                            0,
-                            arr.shape[axis] + div * np.random.randint(1, 10),
-                            div,
-                        )
-                    )
-                )
-                indices_arr.append(
-                    list(
-                        range(
-                            arr.shape[axis] + div * np.random.randint(1, 10),
-                            0,
-                            -div,
-                        )
-                    )
-                )
-
-        # if indivisible
-        if len(indices_arr) == 0:
-            indices_arr.append(1)
-            indices_arr.append(arr.shape[axis])
-            indices_arr.append(list(range(1, arr.shape[axis], 1)))
-
-            # out index
-            indices_arr.append(
-                list(
-                    range(0, arr.shape[axis] + 1 * np.random.randint(1, 10), 1)
-                )
-            )
-            indices_arr.append(
-                list(
-                    range(
-                        arr.shape[axis] + 1 * np.random.randint(1, 10), 0, -1
-                    )
-                )
-            )
+    elif axis_size > 1:
+        for div in range(2, int(math.sqrt(axis_size) + 1)):
+            if axis_size % div == 0:
+                indices_arr.append(div)  # add divisible integer
+                even_div = div
+
+        # add 1 and axis_size
+        indices_arr.append(1)
+        indices_arr.append(axis_size)
+
+    # an index in the dimension of the array along axis
+    indices_arr.append(list(range(1, axis_size, even_div)))
+
+    # an index exceeds the dimension of the array along axis
+    indices_arr.append(
+        list(range(0, axis_size + even_div * random_integer, even_div))
+    )
+    indices_arr.append(
+        list(range(axis_size + even_div * random_integer, 0, -even_div))
+    )
 
     return indices_arr
 
 
 @pytest.mark.parametrize("size", SIZES, ids=str)
 def test_split(size):
     a = np.random.randint(low=0, high=100, size=size)
-    axis_list = list(range(a.ndim))
-    axis_list.append(-1)
+    axis_list = list(range(a.ndim)) + [-1]
     for axis in axis_list:
         input_arr = get_indices(a, axis)
-
         for input_opt in input_arr:
-            func_num = getattr(num, "split")
-            func_np = getattr(np, "split")
-
-            res_num = func_num(a, input_opt, axis=axis)
-            res_np = func_np(a, input_opt, axis=axis)
-
+            res_num = num.split(a, input_opt, axis=axis)
+            res_np = np.split(a, input_opt, axis=axis)
             assert compare_array(res_num, res_np)
 
 
-@pytest.mark.parametrize("func_name", ARG_FUNCS)
-@pytest.mark.parametrize("size", SIZES_NO_EMPTY, ids=str)
-def test_split_different_split(size, func_name):
+@pytest.mark.parametrize("size", SIZES_HSPLIT, ids=str)
+def test_vsplit(size):
```

Comment:
Should this be `SIZES_VSPLIT` for `test_vsplit`?

Reply:
fix